bin/docker-to-squash.py

```diff
@@ -0,0 +1,1430 @@
+                                     hdfs_root, replication,
+                                     image_tag_to_hash_hash)
+  finally:
+    if working_dir:
+      if os.path.isdir(working_dir):
+        shutil.rmtree(working_dir)
+
+def copy_update(args):
+  image_tag_to_hash = args.image_tag_to_hash
+  replication = args.replication
+  force = args.force
+  src_root = args.src_root
+  dest_root = args.dest_root
+  images_and_tags = args.images_and_tags
+  bootstrap = args.bootstrap
+
+  src_layers_dir = src_root + "/layers"
+  src_config_dir = src_root + "/config"
+  src_manifest_dir = src_root  + "/manifests"
+  dest_layers_dir = dest_root + "/layers"
+  dest_config_dir = dest_root + "/config"
+  dest_manifest_dir = dest_root  + "/manifests"
+
+  if bootstrap:
+    hdfs_dirs = [dest_root, dest_layers_dir, dest_config_dir, dest_manifest_dir]
+    image_tag_to_hash_path = hdfs_root + "/" + image_tag_to_hash
+    setup_squashfs_hdfs_dirs(hdfs_dirs, image_tag_to_hash_path)
+  working_dir = None
+
+  try:
+    working_dir = get_working_dir(args.working_dir)
+    local_src_image_tag_to_hash = os.path.join(working_dir, "src-"
+                                               + os.path.basename(image_tag_to_hash))
+    local_dest_image_tag_to_hash = os.path.join(working_dir, "dest-"
+                                                + os.path.basename(image_tag_to_hash))
+
+    src_hash_to_tags, src_tag_to_hash, src_image_tag_to_hash_hash = populate_tag_dicts(src_root, image_tag_to_hash, local_src_image_tag_to_hash)
+    dest_hash_to_tags, dest_tag_to_hash, dest_image_tag_to_hash_hash = populate_tag_dicts(dest_root, image_tag_to_hash, local_dest_image_tag_to_hash)
+
+    for image_and_tag_arg in images_and_tags:
+      image, tags = split_image_and_tag(image_and_tag_arg)
+      if not image:
+        raise Exception("Positional parameter requires an image: " + image_and_tag_arg)
+      if not tags:
+        logging.debug("Tag not given. Using image tag instead: %s", image)
+        tags = [image]
+
+      src_manifest_hash = src_tag_to_hash.get(image, None)
+      if not src_manifest_hash:
+        logging.info("Manifest not found for image %s. Skipping", image)
+        continue
+
+      src_manifest_path = src_manifest_dir + "/" + src_manifest_hash
+      dest_manifest_path = dest_manifest_dir + "/" + src_manifest_hash
+      src_manifest, src_manifest_hash = get_hdfs_manifest_from_path(src_manifest_path)
+
+      src_config_hash = get_config_hash_from_manifest(src_manifest)
+      src_config_path = src_config_dir + "/" + src_config_hash
+      dest_config_path = dest_config_dir + "/" + src_config_hash
+
+      src_layers = get_layer_hashes_from_manifest(src_manifest)
+      dest_layers_paths = [dest_layers_dir + "/" + layer + ".sqsh" for layer in src_layers]
+
+      logging.debug("Copying Manifest: %s", str(src_manifest_path))
+      logging.debug("Copying Layers: %s", str(src_layers_paths))
```

Comment:
src_layers_paths is used here and on 1104, but is not defined/assigned anywhere.

Reply:
missed 
```
src_layers_paths = [src_layers_dir + "/" + layer + ".sqsh" for layer in src_layers]
```

Fixed and tested copy_update